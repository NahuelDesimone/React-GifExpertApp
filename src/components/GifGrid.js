import React, { useEffect, useState } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';



export const GifGrid = ({ category }) => {
    
    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className='card animate__animated animate__fadeIn'>{category}</h3>

            {loading && <p className='card animate__animated animate__flash'>Loading...</p>}

            {<div className='card-grid'>
                <ol>
                    {
                        images.map((img) => (
                            <GifGridItem
                                key={img.id}
                                {...img} />
                        ))
                    }

                </ol>

            </div>}
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


