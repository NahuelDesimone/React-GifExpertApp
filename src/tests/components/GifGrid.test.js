import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //Con el mock voy a fingir cualquier llamada a useFetchGifs

describe('Pruebas en el componente GifGrid', () => {

    const category = "Futbol";

    test('debe de hacer un snapshot con el componente GifGrid', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGfis', () => {
      
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });   
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        
    });
    
    
  
});

