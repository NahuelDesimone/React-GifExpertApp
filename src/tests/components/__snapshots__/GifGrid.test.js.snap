// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid debe de hacer un snapshot con el componente GifGrid 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    Futbol
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid debe de mostrar items cuando se cargan imagenes useFetchGfis 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    Futbol
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Cualquier cosa"
        url="https://localhost/cualquier/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;
